api/catalog/management/commands/resendoauthverification.py

```diff
@@ -0,0 +1,225 @@
+            "--dry_run",
+            help="Count the records that will be removed but don't apply any changes.",
+            type=bool,
+            default=True,
+            action=argparse.BooleanOptionalAction,
+        )
+
+    @transaction.atomic
+    def _handle_email(self, email, dry):
+        """
+        1. Get all application IDs for the email
+        2. Use the one with the lowest ID as the "original" attempt
+        3. Delete the rest
+        4. Delete OAuth2Registrations for the email not associated
+            with the "original" application
+        5. Delete OAuth2Verifications for the email not associated
+            with the "original" application
+
+        This ignores the fact that someone could have tried to register
+        multiple unverified but distinct applications under the same email.
+        This is unlikely given that none of the requests would have worked
+        and that the "feature" isn't explicitly documented anyway.
+        """
+        application_ids = list(
+            OAuth2Registration.objects.filter(email=email)
+            .select_related("associated_application")
+            .order_by("id")
+            .values_list("pk", flat=True)
+        )
+
+        application_to_verify = ThrottledApplication.objects.get(pk=application_ids[0])
+
+        deleted_applications = 0
+        deleted_registrations = 0
+        deleted_verifications = 0
+        if len(application_ids) > 1:
+            applications_to_delete_ids = application_ids[1:]
+            deleted_applications = len(applications_to_delete_ids)
+            if not dry:
+                ThrottledApplication.objects.filter(
+                    pk__in=applications_to_delete_ids
+                ).delete()
+
+            registrations_to_delete = OAuth2Registration.objects.filter(
+                email=email
+            ).exclude(name=application_to_verify.name)
+            deleted_registrations = registrations_to_delete.count()
+            if not dry:
+                registrations_to_delete.delete()
+
+            verifications_to_delete = OAuth2Verification.objects.filter(
+                email=email
+            ).exclude(associated_application=application_to_verify)
+            deleted_verifications = verifications_to_delete.count()
+            if not dry:
+                verifications_to_delete.delete()
+
+        if not dry:
+            verification = OAuth2Verification.objects.get(
+                associated_application=application_to_verify
+            )
+            token = verification.code
+            # We don't have access to `request.build_absolute_uri` so we
+            # have to build it ourselves for the production endpoint
+            link = (
+                f"https://api.openverse.engineering/{reverse('verify-email', [token])}"
+            )
+            verification_msg = verification_msg_template.format(link=link)
+            send_mail(
+                subject="Verify your API credentials",
```

Comment:
Maybe we should make this more specific so folks are inclined to click through?
```suggestion
                subject="Verify your Openverse API credentials",
```

Reply:
Yes! When I commit your suggestion I'll update it in the regularly sent email as well.